Functions/ImageProcessing/xASL_im_ASLSubtractionAveraging.m

```diff
@@ -419,12 +413,112 @@
+	    else
+		    x.Q.T2 = 85;
+		    fprintf('%s\n',['Warning: Unknown T2 for ' num2str(x.MagneticFieldStrength) 'T scanners, using 3T value']);
+	    end
+    end
+
+    if ~isfield(x.Q,'BloodT1') || isempty(x.Q.BloodT1)
+	    % T1 relaxation time of arterial blood
+        % There are 3 options for x.Q.BloodT1:
+        % A) users have provided x.Q.BloodT1
+        % B) users have provided x.Hematocrit which is converted to x.Q.BloodT1 above
+        % C) it doesn't exist and is defaulted here based on MagneticFieldStrength
+	    switch(x.MagneticFieldStrength)
+		    case 0.2 
+			    x.Q.BloodT1 = 776; % Rooney 2007 MRM
+                fprintf('%s\n', 'Defaulting x.Q.BloodT1 to 776 ms for 0.2T (Rooney 2007 MRM)');
+		    case 1
+			    x.Q.BloodT1 = 1350; % Rooney 2007 MRM
+                fprintf('%s\n', 'Defaulting x.Q.BloodT1 to 1350 ms for 1T (Rooney 2007 MRM)');
+		    case 1.5
+			    x.Q.BloodT1 = 1540; % Rooney 2007 MRM
+                fprintf('%s\n', 'Defaulting x.Q.BloodT1 to 1540 ms for 1.5T (Rooney 2007 MRM)');
+		    case 3
+			    x.Q.BloodT1 = 1650; % Alsop 2015 MRM
+                fprintf('%s\n', 'Defaulting x.Q.BloodT1 to 1650 ms for 3T (Alsop 2015 MRM)');
+		    case 4
+			    x.Q.BloodT1 = 1914; % Rooney 2007
+                fprintf('%s\n', 'Defaulting x.Q.BloodT1 to 1914 ms for 4T (Rooney 2007 MRM)');
+		    case 7
+			    %x.Q.BloodT1 = 2578; % Rooney 2007 MRM
+			    x.Q.BloodT1 = 2100; % Ivanov 2017 NeuroImage
+                fprintf('%s\n', 'Defaulting x.Q.BloodT1 to 2100 ms for 7T (Ivanov 2007 NeuroImage)');
+		    otherwise
+			    x.Q.BloodT1 = 1650; % Alsop 2015 MRM - assuming default 3 T
+			    fprintf('%s\n',['Warning: Unknown T1-blood for ' xASL_num2str(x.MagneticFieldStrength) 'T scanner, using 3T value (Alsop 2015 MRM)']);
+                % PM: NOTE that this situation is unlikely, given that we
+                % default to x.MagneticFieldStrength = 3 at section 0
+                % Administration above
+        end
+    end
+    %% =================================================================================
+    %% PM: THIS PART ON T2STAR & T2 CAN BE ADDED TO THE START OF xASL_module_ASL INSTEAD    
+
+
+
+
+    % EchoTime weighting
+    % We default to T2*, for 3D GRASE and 2D EPI.
+    % If we detect 3D spiral, we go for T2 instead.
+    % For most 3D spiral sequences (==GE), we do not have individual control-label pairs anyway.
+
+    if isfield(x.Q, 'Sequence') && ~isempty(regexpi(x.Q.Sequence, '3d_spiral'))
+        T2_factor = x.Q.T2;
+    else
+        T2_factor = x.Q.T2star;
+    end
+
+    nsrTE = exp(x.Q.EchoTime_PWI3D ./T2_factor); % for each volume, get the EchoTime weighting
+
+    % PLD & LD weighting
+    nsrPLD = exp(x.Q.InitialPLD_PWI3D ./ x.Q.BloodT1) ./ (1-exp(-x.Q.LabelingDuration_PWI3D ./x.Q.BloodT1));
+
+    % Perfusion-weighting (PW) vs vascular weighting
+    pwPLD = max((min(x.Q.InitialPLD_PWI3D, 2500) - 1000) ./ 15, 1) ./100;
```

Comment:
I disagree. The sum of contribution should be 1, so you want a fraction for each volume. See my comments as examples. I know it doesn't matter mathematically but nicer to follow perhaps.

Fixed in 6fda911643cea1f2783ea578410f3f6f5f94beae. Do you agree?

Reply:
Exactly - mathematically it doesn't matter. Practically, people will be confused why there's a superfluous division by 100 that makes no difference.

You are calculating a weighted sum - there's no justification for having numbers <1.

I don't care that much, but you usually want to have things clean and explained ;) I'm just saying that this explanation doesn't really show much logic to me

Reply:
To be clear - I don't care about the division, I was just raising this point because you usually care. 
I know that they are not programmers - but especially for this reason, they will be confused that you have something in percentages, something between 0-1 and something as nsrPLD has no units at all. 

So somebody can be totally confused why are you calculating percentages for pwPLD and why.

But really no issue for me - I don't care about programmers or mathematicians. I was just saying that a normal simple user will not understand why do you have to divide by hundred and convert to percentages. If you think it is better for normal users, then I'm perfectly fine with it. Your decision really. My main point was - not necessary and potentially confusing, but you can keep it as far as I'm concerned as it has no influence.